### ISSUE TITLE
SPF macros not evaluated

### ISSUE THREAD

**Issue by blade5502** [MailCleaner/MailCleaner#379]
Hi,

the SPF checker doesn't evaluate macros resulting in spf fails instead of passes. As they are standardized in the SPF RFC's 4408 and 7208 they should be evaluated by SPF checker. The only workaround for now is to whitelist domains with such a record.

Example of a SPF record with macros: `v=spf1 exists:%{i}._i.%{d}._d.espf.agari.com include:%{d}.33.spf-protect.agari.com -all
`

Here is a good description of SPF macros with examples: https://www.jamieweb.net/blog/using-spf-macros-to-solve-the-operational-challenges-of-spf/

**Comment by JohnMertz** [comment]
LibSPF2 was updated with Exim 4.96 earlier this year. I just configured my personal domain to use only a macro in the SPF record:

```
$ host -t txt john.me.tz
john.me.tz descriptive text "v=spf1 exists:%{i}.john.me.tz -all"
$ host 144.217.12.101.john.me.tz
144.217.12.101.john.me.tz has address 127.0.0.2
```

and it appears that it passes on a MailCleaner appliance running Exim 4.96. Please confirm that this is no longer an issue for you.

**Comment by KlugFR** [comment]
Our appliances were updated during summer (I saw the libSPF2 message in the infobox).
We're still having issues with macros.

However, thanks to your message I did a bit of digging.

SPF record for the domain that showed up with the issue: 
```
cnav.fr descriptive text "v=spf1 ip4:90.85.58.17 ip4:83.118.202.17 ip4:90.85.58.55 ip4:83.118.202.55 ip4:193.168.50.136 ip4:193.168.50.140 ip4:93.94.162.10 include:spf.protection.outlook.com include:spf.rp01.net exists:%{i}.spf.hc898-42.eu.iphmx.com -all"
```

Mail sent from: 23.90.111.238

Host resolution for this IP from the mailcleaner appliance that rejected the mail:
```
$ host 23.90.111.238.spf.hc898-42.eu.iphmx.com
$ 
```

Host resolution for this IP from my laptop:
```
$ host 23.90.111.238.spf.hc898-42.eu.iphmx.com
23.90.111.238.spf.hc898-42.eu.iphmx.com has address 127.0.0.2
```

Looks like the issue might be the resolver used (OpnSense appliance).
There's a "DNS rebinding" security activated (see: https://forum.opnsense.org/index.php?topic=18503.0 ).
Once disabled, I'm getting the correct answer.

Looks like the issue is fixed for me too.
